src/Websites/Websites/Cmdlets/Certificates/NewAzureWebAppCertificate.cs

```diff
@@ -0,0 +1,121 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.WebApps.Models;
+using Microsoft.Azure.Commands.WebApps.Models.WebApp;
+using Microsoft.Azure.Commands.WebApps.Utilities;
+using Microsoft.Azure.Management.WebSites.Models;
+using Microsoft.Rest.Azure;
+using System;
+using System.Linq;
+using System.Management.Automation;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.WebApps.Cmdlets.Certificates
+{
+
+    /// <summary>
+    /// This commandlet will let you create a new managed certificate
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "WebAppCertificate", SupportsShouldProcess = true), OutputType(typeof(PSCertificate))]
+    public class NewAzureWebAppCertificate : WebAppBaseClientCmdLet
```

Comment:
I still don't see the optional parameter for certificate Name - so users can provide their own name for certificate instead of the name we generate.

Reply:
As I said in my last reply from your previous review, the name/canonical name should be same as Hostname. it is not allowing the other names than the hostname.